frontend/components/auth/UnlockKeyringDialog.tsx

```diff
@@ -46,23 +46,24 @@ export default function UnlockKeyringDialog(props: { organisation: OrganisationT
   const decryptKeyring = (sudoPassword: string) => {
     return new Promise(async (resolve, reject) => {
       setUnlocking(true)
+      setTimeout(async() => {
+        try {
+          if (trustDevice) {
+            setDevicePassword(organisation.memberId!, sudoPassword)
+          }
 
-      try {
-        if (trustDevice) {
-          setDevicePassword(organisation.memberId!, sudoPassword)
-        }
-
-        const decryptedKeyring = await getKeyring(session?.user?.email!, organisation, sudoPassword)
-        setKeyring(decryptedKeyring)
-        setUnlocking(false)
-        reset()
-        closeModal()
-        resolve(true) // Resolve the promise successfully
-      } catch (e) {
-        console.error(e)
-        setUnlocking(false)
-        reject(e) // Reject the promise with the error
-      }
+          const decryptedKeyring = await getKeyring(session?.user?.email!, organisation, sudoPassword)
+          setKeyring(decryptedKeyring)
+          setUnlocking(false)
+          reset()
+          closeModal()
+          resolve(true) // Resolve the promise successfully
+        } catch (e) {
+          console.error(e)
+          setUnlocking(false)
+          reject(e) // Reject the promise with the error
+        }  
+      }, 2000);
```

Comment:
@4rjunc Is it possible to use a smaller delay? 2 seconds seems like a lot. How about ~300ms?

Reply:
Okay will change it.

Reply:
Pushed the changes.

Reply:
@4rjunc One side effect of this change is that even when the device is trusted, this delay causes the popup to show for much longer than before: 

[Screencast from 07-27-2024 10:26:04 PM.webm](https://github.com/user-attachments/assets/38ec6302-8ac7-4dfc-9c96-91dca49f49cb)


I'd like to try and reduce the timeout delay as much as possible. I've tested `100`ms delay locally and it seems to work fine. Could you try that on your system as well and see if it works correctly? If that works, then lets go ahead with that.

Reply:
<img width="771" alt="Screenshot 2024-07-27 at 23 48 47" src="https://github.com/user-attachments/assets/528d87b5-52f8-4b32-9a55-4212ac8b2ba4">
When the device is trusted , `trustDevice` variable is false. I got confused. I have checked with 100ms it works fine.

Reply:
Great! Could you push that change? We can get this merged.

Reply:
I couldn't get the fix for the time lag issue when the device is trusted. I have reduced the timeout from 300 -> 100